src/test/util/setup_common.cpp

```diff
@@ -234,21 +234,38 @@ void TestChain100Setup::mineBlocks(int num_blocks)
     }
 }
 
-CBlock TestChain100Setup::CreateAndProcessBlock(const std::vector<CMutableTransaction>& txns, const CScript& scriptPubKey)
+CBlock TestChain100Setup::CreateBlock(
+    const std::vector<CMutableTransaction>& txns,
+    const CScript& scriptPubKey,
+    CChainState* chainstate)
```

Comment:
In commit "test: refactor: separate CreateBlock in TestChain100Setup" (6d5d09634ac354c16c298c83d7b699b2147a98b3)

Would be nice to make the last argument a `CChainState&` reference instead of a pointer since it isn't allowed to be null (and it's a new method so there's no backward compatility concern)

Reply:
In commit "test: refactor: separate CreateBlock in TestChain100Setup" (e061837697ed8ebf8d5443478829b48f3fe0e8db)

> Would be nice to make the last argument a `CChainState&` reference instead of a pointer since it isn't allowed to be null (and it's a new method so there's no backward compatility concern)

(This is still applicable, in case the PR gets another update)